File: Greedy/11.Container_With_Most_Water/Container_With_Most_Water.py
class Solution:
    def maxArea(self, h: List[int]) -> int:
        mx_water = 0
        l = 0; r = len(h) - 1 
        while(l < r):
            mx_water = max(mx_water, (r - l) * min(h[l], h[r]))
            if h[l] < h[r]: l += 1
            else: r -= 1
        return mx_water        
        
        
        

File: Greedy/11.Container_With_Most_Water/Container_With_Most_Water.java
class Solution {
    public int maxArea(int[] h) {
        int mx_water = 0;
        int l = 0, r = h.length - 1;
        while(l < r){
            int curr_water = (r - l) * Math.min(h[l], h[r]);
            mx_water = Math.max(curr_water, mx_water);
            if(h[l] < h[r]) l++;
            else r--;
        }
        return mx_water;
    }
}